src/sinks/datadog/metrics/config.rs

```diff
@@ -127,6 +133,30 @@ impl DatadogMetricsEndpointConfiguration {
     }
 }
 
+/// Payload limits for metrics are endpoint-dependent.
+pub(super) struct DatadogMetricsPayloadLimits {
+    pub(super) uncompressed: usize,
+    pub(super) compressed: usize,
+}
+
+impl From<DatadogMetricsEndpoint> for DatadogMetricsPayloadLimits {
+    fn from(endpoint: DatadogMetricsEndpoint) -> Self {
```

Comment:
Is there any benefit to making this an `impl From` and not just a plain method:
```suggestion
impl DatadogMetricsEndpoint {
    const fn payload_limits(&self) -> DatadogMetricsPayloadLimits {
```

Reply:
Just a plain method makes more sense